workspace/apps/pidp/src/app/features/organization-info/pages/endorsements/endorsements.page.ts

```diff
@@ -315,4 +316,25 @@ export class EndorsementsPage
       }),
     );
   }
+
+  public getStatus(endorsementRequestStatus: EndorsementRequestStatus): string {
+    let statusText = 'Requested';
+    if (endorsementRequestStatus === EndorsementRequestStatus.CANCELLED) {
+      statusText = 'Cancelled';
+    } else if (endorsementRequestStatus === EndorsementRequestStatus.DECLINED) {
+      statusText = 'Declined';
+    } else if (endorsementRequestStatus === EndorsementRequestStatus.APPROVED) {
+      statusText = 'In progress';
+    }
+    return statusText;
+  }
+
+  public isEndorsementRequested(
+    endorsementRequestStatus: EndorsementRequestStatus,
+  ): boolean {
+    return (
+      endorsementRequestStatus !== EndorsementRequestStatus.CANCELLED &&
```

Comment:
Rather than checking the conditions independently for each color define a constant array in constants like 

`const red_color_status_text = [ EndorsementRequestStatus.CANCELLED,  EndorsementRequestStatus.DECLINED];`

Import this array into current component.
Then current function will be 

`
 public isEndorsementRequested(
    endorsementRequestStatus: EndorsementRequestStatus,
  ): boolean {
return !red_color_status_text.contains(endorsementRequestStatus);
}
`